src/shared/ui/Input.tsx

```diff
@@ -1,20 +1,23 @@
 import {FieldPath, UseFormRegisterReturn} from "react-hook-form";
 import {useState} from "react";
-import {ILoginFormValues, IRegisterFormValues} from "@/shared/types";
+import {IGlobalFilterFormValues, ILoginFormValues, IRegisterFormValues} from "@/shared/types";
 
 interface IInputProps {
   onChange?: () => void;
   HTMLType: 'email' | 'text' | 'password';
   iconType?: 'person' | 'mail' | 'password' | 'search-icon-gray';
   value?: string;
   placeholder?: string;
-  extraClass?: string;
+  extraBoxClass?: string;
+  extraInputClass?: string;
   defaultValue?: string;
   error?: any;
-  hookFormValues?: UseFormRegisterReturn<FieldPath<ILoginFormValues>> | UseFormRegisterReturn<FieldPath<IRegisterFormValues>>;
+  hookFormValues?: UseFormRegisterReturn<FieldPath<ILoginFormValues>> | UseFormRegisterReturn<FieldPath<IRegisterFormValues>> | UseFormRegisterReturn<FieldPath<IGlobalFilterFormValues>>;
+  onInput?: any;
+  inputValue: string;
```

Comment:
onChange is more conventional in this context
the same with inputValue, since our component is **Input**  better to call the pop just value